core/src/main/java/org/elasticsearch/common/util/concurrent/PrioritizedEsThreadPoolExecutor.java

```diff
@@ -175,6 +177,7 @@ public Pending(Object task, Priority priority, long insertionOrder, boolean exec
 
         @Override
         public void run() {
+            started = true;
             FutureUtils.cancel(timeoutFuture);
```

Comment:
It would have helped me a lot to understand the logic if you added a comment here saying that `timeoutFuture` might not have been initialized here yet and because of this we have to try cleaning it here and in `scheduleTimeout`.